### PR TITLE
[FLINK-20883][table-planner-blink] Separate the implementation of BatchExecOverAggregate and StreamExecOverAggregate

### PR DESCRIPTION
**Pull request by godfreyhe**
## What is the purpose of the change

*Separate the implementation of BatchExecOverAggregate and StreamExecOverAggregate*

## Brief change log

  - *Introduce StreamPhysicalOverAggregate, and make StreamExecOverAggregate only extended from ExecNode*
  - *Introduce StreamPhysicalPythonOverAggregate, and make StreamExecPythonOverAggregate only extended from ExecNode*
  - *Introduce BatchPhysicalOverAggregate, and make BatchExecOverAggregate only extended from ExecNode*
  - *Introduce BatchPhysicalPythonOverAggregate, and make BatchExecPythonOverAggregate only extended from ExecNode*

## Verifying this change

This change is a refactoring rework covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/over/MultiFieldRangeBoundComparatorCodeGenerator.scala [hunk 1]
```diff
@@ -48,9 +46,14 @@ class MultiFieldRangeBoundComparatorCodeGenerator(
     }
 
     val ctx = CodeGeneratorContext(conf)
-
     val compareCode = GenerateUtils.generateRowCompare(
```
Comment (wenlong88) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/over/MultiFieldRangeBoundComparatorCodeGenerator.scala:49]: use the new method accepts sortSpec

**Review by wenlong88 — COMMENTED** [review]
the overall change LGTM, just left some minor comments and OverWindowTest is missed in the renaming.

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecOverAggregate.java [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+        final List<LogicalType> fieldTypes = new ArrayList<>(inputRowType.getChildren());
+        IntStream.range(0, constants.size()).forEach(i -> fieldNames.add("TMP" + i));
+        for (int i = 0; i < constants.size(); ++i) {
+            fieldNames.add("TMP" + i);
+            fieldTypes.add(FlinkTypeFactory.toLogicalType(constants.get(i).getType()));
+        }
+
+        final RowType aggInputRowType =
+                RowType.of(
+                        fieldTypes.toArray(new LogicalType[0]), fieldNames.toArray(new String[0]));
```
[the remark this answers is not in the document]
Comment (godfreyhe) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecOverAggregate.java:140]: I think it's easy for debug

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchPhysicalOverAggregateBase.scala [hunk 3]
```diff
@@ -18,88 +18,50 @@
...
-    group.lowerBound.isUnbounded && !group.upperBound.isUnbounded
-
-  protected def isUnboundedFollowingWindow(group: Window.Group) =
-    !group.lowerBound.isUnbounded && group.upperBound.isUnbounded
-
-  protected def isSlidingWindow(group: Window.Group) =
-    !group.lowerBound.isUnbounded && !group.upperBound.isUnbounded
-
-  def getGrouping: Array[Int] = grouping
+  protected lazy val newWindowGroups: Seq[Group] = splitOutOffsetOrInsensitiveGroup(windowGroups)
```
[the remark this answers is not in the document]
Comment (godfreyhe) [reply flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchPhysicalOverAggregateBase.scala:64]: how about `offsetAndInsensitiveSensitiveGroups`

**Review by wenlong88 — APPROVED** [review]
LGTM.

**Comment by godfreyhe** [comment]
Thanks for the review, I will fix the conflicts in my local and merge the pr

**Pull request closed without merging**